odilo.py

```diff
@@ -0,0 +1,581 @@
+        """Fetch a new Bearer Token and update the given Credential object."""
+        try:
+            response = self.token_post(
+                self.TOKEN_ENDPOINT,
+                dict(grant_type="client_credentials"),
+                allowed_response_codes=[200]
+            )
+            data = response.json()
+        except (HTTPException, RequestNetworkException) as e:
+            self.log.error("Cannot connect with %s, error: %s" % (self.TOKEN_ENDPOINT, e.message))
+            return None
+
+        if response.status_code == 200:
+            self._update_credential(credential, data)
+            self.token = credential.credential
+            return 'OK'
+        elif response.status_code == 400:
+            response = response.json()
+            message = response['error']
+            if 'error_description' in response:
+                message += '/' + response['error_description']
+            return 'message'
+
+    def get(self, url, extra_headers={}, exception_on_401=False):
+        """Make an HTTP GET request using the active Bearer Token."""
+        if extra_headers is None:
+            extra_headers = {}
+        headers = dict(Authorization="Bearer %s" % self.token)
+        headers.update(extra_headers)
+        status_code, headers, content = self._do_get(url, headers)
+        if status_code == 401:
+            if exception_on_401:
+                # This is our second try. Give up.
+                raise BadResponseException.from_response(
+                    url,
+                    "Something's wrong with the Odilo OAuth Bearer Token!",
+                    (status_code, headers, content)
+                )
+            else:
+                # Refresh the token and try again.
+                self.check_creds(True)
+                return self.get(url, extra_headers, True)
+        else:
+            return status_code, headers, content
+
+    def token_post(self, url, payload, headers={}, **kwargs):
+        """Make an HTTP POST request for purposes of getting an OAuth token."""
+        s = "%s:%s" % (self.client_key, self.client_secret)
+        auth = base64.standard_b64encode(s).strip()
+        headers = dict(headers)
+        headers['Authorization'] = "Basic %s" % auth
+        headers['Content-Type'] = "application/x-www-form-urlencoded"
+        return self._do_post(url, payload, headers, **kwargs)
+
+    @staticmethod
+    def _update_credential(credential, odilo_data):
+        """Copy Odilo OAuth data into a Credential object."""
+        credential.credential = odilo_data['token']
+        expires_in = (odilo_data['expiresIn'] * 0.9)
+        credential.expires = datetime.datetime.utcnow() + datetime.timedelta(seconds=expires_in)
+
+    def get_metadata(self, record_id):
+        identifier = record_id
+        if isinstance(record_id, Identifier):
+            identifier = record_id.identifier
+
+        url = self.RECORD_METADATA_ENDPOINT.replace('{recordId}', identifier)
```

Comment:
I expected you to use `string.format` here. Either change this to use `string.format(recordId=identifier)`, or use "%(recordId)s" instead of "{recordId} in your strings and format them with the `%` operator. The difference is cosmetic, but in most other places in the app we use the "%(recordId)s" style.

Reply:
Ok, I have changed all string.replace for string.format